src/main/kotlin/com/jetbrains/snakecharm/lang/psi/SmkFileReference.kt

```diff
@@ -0,0 +1,31 @@
+package com.jetbrains.snakecharm.lang.psi
+
+import com.intellij.openapi.util.TextRange
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VirtualFileManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiManager
+import com.intellij.psi.PsiReferenceBase
+import com.jetbrains.python.psi.PyStringLiteralExpression
+
+class SmkFileReference(
+        element: PsiElement, textRange: TextRange
+) : PsiReferenceBase<PsiElement>(element, textRange) {
+    override fun resolve(): PsiElement? {
+        if (element !is PyStringLiteralExpression) {
+            return null
+        }
+
+        val parentFolder = element.containingFile.virtualFile?.parent ?: return null
+
+        val filePath = (element as PyStringLiteralExpression).stringValue
+        val vfm = VirtualFileManager.getInstance()
+
+        // Trying to find the file anywhere
+        val virtualFile = parentFolder.findFileByRelativePath(filePath) ?:
+        vfm.getFileSystem(LocalFileSystem.PROTOCOL).findFileByPath(filePath) ?:
+        vfm.findFileByUrl(filePath)
+
+        return virtualFile?.let { PsiManager.getInstance(element.project).findFile(it) }
+    }
```

Comment:
What about 'getVariants` ? Normally resolve() and getVariants() are more or less consistent and implmented together. E.g. in case of `include` we could provided references to all snakemake files in current module or project (except current file). For `config` - *.yaml files, for `workdir` - probably file folders. Maybe such file path reference is already implemented in IntelliJ Platform.

Reply:
P.S: You could file a separate issue for code completion here